### ISSUE TITLE
js.lib.AsyncIterator is redefined

### ISSUE THREAD

**Issue by andyli** [haxiomic/dts2hx#98]
* Haxe 4.2.3
 * dts2hx 0.16.0
* `@types/node` 16.10.1

build.hxml
```hxml
--class-path lib
--js dummy.js
--no-output
--macro include("global")
--macro include("js")
--macro include("node")
```

To reproduce,
```sh
$ npx dts2hx @types/node --noLibWrap --output lib
$ haxe build.hxml
(unknown) : Type name js.lib.AsyncIterator is redefined from module js.lib.AsyncIterator
```

The cause is Haxe std lib defined `js.lib.AsyncIterator` in [`std/js/lib/Iterator.hx`](https://github.com/HaxeFoundation/haxe/blob/1385eda48e60387e349282d26347b28d0b2768ca/std/js/lib/Iterator.hx#L39), and somehow dts2hx generates another `AsyncIterator`.

**Comment by haxiomic** [comment]
Still thinking on best resolution, but talking through the problem:

- AsyncIterator was added to haxe more recently, so you can use `--useSystemHaxe` and it'll map to the std lib of the currently install haxe (by default it uses stdlib for 4.0.5 but maybe --useSystemHaxe should be default)

- However, enabling this doesn't solve the issue, because the haxe std lib extern doesn't have all the type parameters required:
```haxe
typedef AsyncIterator<T, TReturn, TNext> = {}
```
vs
```
typedef AsyncIterator<T> = {
	function next():Promise<IteratorStep<T>>;
}
```

So we probably have to generate an AsyncIterator anyway, but it's probably best to rename it to avoid the clash

**Comment by haxiomic** [comment]
This is fixed in 0.16.1, we still generate our version of `AsyncIterator` because other types rely on the type parameters being correct but it's renamed to `AsyncIterator_`

Running your test build.hxml now passes for me, but I changed `--macro include("js")` to `--macro include("js.lib")` to avoid compiler issues with HxOverrides.hx

**Comment by andyli** [comment]
Works perfectly. Thanks!
